middle_end/flambda2/to_cmm/to_cmm_primitive.ml

```diff
@@ -658,7 +658,10 @@ let nullary_primitive _env res dbg prim =
        `to_cmm_shared.ml` *)
     let expr = Cmm.Cop (Cprobe_is_enabled { name }, [], dbg) in
     None, res, expr
-  | Begin_region | Begin_try_region -> None, res, C.beginregion ~dbg
+  | Begin_region { ghost = false } | Begin_try_region { ghost = false } ->
+    None, res, C.beginregion ~dbg
+  | Begin_region { ghost = true } | Begin_try_region { ghost = true } ->
+    None, res, C.unit ~dbg
```

Comment:
I'm not sure unit is the correct thing here. Considering that it is supposed to return a region, something along the lines of 1 would be better I'd think. That being said, we could also ignore/erase ghost regions at that point if that is the goal.

Reply:
I think it's safe, but I'd still expect the cmm expression returned to, at least morally, be something of the correct type/kind for a region. In the case of regions, that means a pointer (usually that points inside the local allocation stack), so I think returning the null pointer / `0` would be better here.

Alternatively, this case could simply raise (at compilation time) if we want to guarantee that these regions are correctly eliminated (but that would probably require changes to `to_cmm_expr` to not try and produce cmm code for these primitives).

Reply:
I've changed this to use `int 0`, that should be fine for now.